docs/template_plugin/tests/functional/op_reference/scatter_update.cpp

```diff
@@ -0,0 +1,914 @@
+                                                                  19, 20,
+                                                                  21, 22,
+                                                                  23, 24,
+                                                                  25, 26,
+                                                                  27, 28,
+                                                                  29, 30,
+                                                                  31, 32}})
+            .axis({{1}, integer_type, std::vector<I> {2}})
+            .expected({{5, 2, 2, 2}, numeric_type, std::vector<N> {
+                                                               1, 2,
+                                                               3, 4,
+                                                               9, 10,
+                                                               11, 12,
+                                                               17, 18,
+                                                               19, 20,
+                                                               25, 26,
+                                                               27, 28,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0,
+                                                               0, 0}})};
+    return ScatterUpdateParams;
+}
+
+std::vector<ScatterUpdate3Params> generateScatterUpdateCombinedParams() {
+    const std::vector<std::vector<ScatterUpdate3Params>> ScatterUpdateTypeParams {
+        // f32
+        generateScatterUpdate3Params<element::Type_t::f32, element::Type_t::i16>(element::f32, element::i16),
+        generateScatterUpdate3Params<element::Type_t::f32, element::Type_t::i32>(element::f32, element::i32),
+        generateScatterUpdate3Params<element::Type_t::f32, element::Type_t::i64>(element::f32, element::i64),
+        // f16
+        generateScatterUpdate3Params<element::Type_t::f16, element::Type_t::i16>(element::f16, element::i16),
+        generateScatterUpdate3Params<element::Type_t::f16, element::Type_t::i32>(element::f16, element::i32),
+        generateScatterUpdate3Params<element::Type_t::f16, element::Type_t::i64>(element::f16, element::i64),
```

Comment:
As I see the Scatter's evaluate doesn't support float types.
Should we extend it?

Reply:
Scatter's evaluate checks the type for the indices input (input number 1) which do not support float types, but the data input support float types. In the similar op ScatterElementsUpdate has_evaluate() method checks both input 0 for data types and input 1 for indices type

Reply:
The `has_evaluate` switch is about second input (indices) type, 
as movement operator ScatterUpdate should work with any first input data type;)
https://github.com/openvinotoolkit/openvino/blob/44fa9310855671d0f31099755e03a0175914fbf7/ngraph/core/src/op/scatter_update.cpp#L98-L107

Reply:
The problem with unsigned integers is that for several test cases we have negative axis values

Reply:
@ilyachur so the only way, that I can think of at the moment,  to cover unsigned integers would be to copy all of the tests without the ones with negative axis values and instantiate TEST_P for the unsigned types